pkg/node/node.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"bytes"
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/klog"
+	azureprovider "k8s.io/legacy-cloud-providers/azure"
+)
+
+// IMDSNodeProvider implements nodemanager.NodeProvider.
+type IMDSNodeProvider struct {
+	azure *azureprovider.Cloud
+}
+
+// NewIMDSNodeProvider creates a new IMDSNodeProvider.
+func NewIMDSNodeProvider() *IMDSNodeProvider {
+	az, err := azureprovider.NewCloud(bytes.NewBuffer([]byte(`{
+			"useInstanceMetadata": true,
+			"vmType": "vmss"
```

Comment:
hardcoded only for `vmss`?

Reply:
yep, vmss vmType support both VMSS and VMAS nodes